dask/array/fft.py

```diff
@@ -28,31 +30,37 @@ def _fft_out_chunks(a, n, axis):
     return chunks
 
 
-def _rfft_out_chunks(a, n, axis):
+def _rfft_out_chunks(a, n, axis, rtype):
     if n is None:
         n = a.chunks[axis][0]
     chunks = list(a.chunks)
-    chunks[axis] = (n // 2 + 1,)
+    if issubclass(np.dtype(rtype).type, numbers.Real):
+        chunks[axis] = (n,)
+    else:
+        chunks[axis] = (n // 2 + 1,)
```

Comment:
I guess, but why? If someone did this, it sort of raises the question of did they understand the purpose of SciPy's API at all. The point was to guarantee real values, keep the same shape, and make it easy to invert later. Making all of the values complex makes the array twice as large, but does not add any value as the imaginary parts will all be zero.

That all said, I can't exactly disagree with you. Trying to handle the SciPy `*rfft` functions is pretty messy and TBH am really not sold on their benefit. They have a few neat tricks, but nothing that really justifies to me why I would want to look at my data in that way instead of an array of complex values.

In short, I'm coming back to the viewpoint that I had when I last looked at this Friday. Namely let's not bother to support the SciPy `*rfft` functions. If someone wants to try later, we have made their job much easier with work done in this PR. Plus we have already opened the door to people wanting the other SciPy FFT functions or using pyFFTW, which is going to help anyone wanting some extra speed to get it easily with Dask.